### PR TITLE
Unittest squash barcodes

### PR DESCRIPTION
**Pull request by sjanssen2**
To check all conditions in the build_barcodes_pdf method, I had to dynamically replace the get_image method. Thus, I needed to change how those methods are imported.

### CODE REVIEW

File: knimin/lib/tests/test_squash_barcodes.py [hunk 1]
```diff
@@ -1,28 +1,68 @@
...
+        # reset function
+        m.get_image = old_get_image
+
+    def test_tmpdir(self):
+        # test that a temporary directory cannot collide with an existing one
+        random.seed(7)
+        fake_tmp_dir = 'tmp_IDQBN'
+        if not path.exists(fake_tmp_dir):
+            mkdir(fake_tmp_dir)
+        self.assertTrue(m.build_barcodes_pdf(['000000011']))
```
Comment (wasade) [inline knimin/lib/tests/test_squash_barcodes.py]: it doesn't look like this test has anything to do with the `tmp_dir`? I'm a bit confused

On a related note, the `tempfile` provides excellent functionality for generating temp files or directories with mechanisms to ensure they're cleaned up if desired

Comment (sjanssen2) [inline knimin/lib/tests/test_squash_barcodes.py]: The `build_barcodes_pdf` method contains the somewhat strange loop `    while path.exists(tmpdir):
        tmpdir = "tmp_%s" % ''.join([choice("ABCDEFGHIJKLMNOPQRSTUVWXYZ")
                                     for i in range(5)])
    mkdir(tmpdir)
` to make sure creating a new, non existing tmp dir. I want to check if this look functions as expected by first creating a colliding dir, such that the loop must be executed at least once.
I agree, it is very confusing.

Comment (wasade) [inline knimin/lib/tests/test_squash_barcodes.py]: "somewhat strange" is polite. it would be great to just leverage python's `tempfile` module

**Comment by wasade** [comment]
Appears there are merge conflicts

**Comment by sjanssen2** [comment]
@wasade : conflicts resolved!

**Pull request merged**